### PR TITLE
docs: #377 update mdbook nodes info

### PR DESCRIPTION
**Pull request by miguelcsx**
## Description

This PR updates the `mdbook` to provide more detailed and accurate information about nodes.

### Changes:
- Revised the nodes information to ensure accuracy and completeness.
- Clarified the usage of the `cli` and detailed the flags related to the Abstract Syntax Tree (AST).

## Motivation and Context

The previous documentation on nodes was incomplete and lacked specific details about `cli` usage and AST flags. These updates aim to improve the comprehensibility and usability of the documentation for users working with nodes and ASTs.

### CODE REVIEW

File: rust-code-analysis-book/src/commands/nodes.md [hunk 1]
```diff
@@ -1,52 +1,45 @@
 # Nodes
 
-**rust-code-analysis-cli** allows to extract some information from the nodes
-which compose the *Abstract Syntax Tree (AST)* of a source code.
+The rust-code-analysis-cli provides commands to analyze and extract information from the nodes in the **AST** of a source file.
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:3]: Why did you remove "Abstract Syntax Tree"?

File: rust-code-analysis-book/src/commands/nodes.md [hunk 2]
```diff
@@ -1,52 +1,45 @@
...
-## Find Errors
+## Error Detection
 
-To know if there are some syntactic errors in your code, run:
+To detect syntactic errors in your code, run:
 
-```console
-rust-code-analysis-cli -p /path/to/your/file/or/directory -I "*.ext" -f -error
+```bash
+rust-code-analysis-cli -p /path/to/file/or/directory -I "*.ext" -f error
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:10]: Let's try to reduce the diff in cases like these.
For example, no need to switch from "console" to "bash", and no need to replace "/path/to/your/file/or/directory" with "/path/to/file/or/directory"

File: rust-code-analysis-book/src/commands/nodes.md [hunk 3]
```diff
@@ -1,52 +1,45 @@
...
-## Print AST
+## Printing the AST
 
-If you want to print the AST of a source code, run the following command:
+To visualize the AST of a source file, use the `--dump` flag:
 
-```console
-rust-code-analysis-cli -p /path/to/your/file/or/directory -d
+```bash
+rust-code-analysis-cli -p /path/to/file/or/directory -I "*.ext" --dump
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:33]: Same here.

Also, I don't see why switching from "-d" to "--dump" here but then keep "-d" in the comment later. Also, I don't see why adding `-I "*.ext"` makes sense.

File: rust-code-analysis-book/src/commands/nodes.md [hunk 4]
```diff
@@ -1,52 +1,45 @@
...
-and `--le` options. The former represents the starting line of the code to be
-considered, while the latter its ending line.
-For example, if we want to print the AST of a single function which starts at
-line 5 and ends at line 10, we need to launch this command:
+To analyze only a specific part of the code, use the `--ls` (line start) and `--le` (line end) options:
 
-```console
-rust-code-analysis-cli -p /path/to/your/file/or/directory -d --ls 5 --le 10
+```bash
+rust-code-analysis-cli -p /path/to/file/or/directory -d --ls 5 --le 10
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:42]: Same here.

Please remove all these changes throughout the PRs, and then let me know when I can re-review them.

**Comment by miguelcsx** [comment]
@marco-c I made the suggested changes, I hope they meet your specifications. :+1:

File: rust-code-analysis-book/src/commands/nodes.md [hunk 5]
```diff
@@ -1,52 +1,45 @@
...
-the language defined by the extension of the file.
-The `-f` option instead searches all nodes of a certain type.
-In the case above, we are looking for all the erroneous nodes present in the
-code.
+- `-p`: Path to a file or directory (analyzes all files in the directory).
+- `-I`: Glob filter for selecting files by extension (e.g., `*.js`, `*.rs`).
+- `-f`: Flag to search for nodes of a specific type (e.g., errors).
 
-## Count Errors
+This command will find and list all syntax errors in the specified code.
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:17]: This sentence can be removed as it's a duplicate of the previous

File: rust-code-analysis-book/src/commands/nodes.md [hunk 6]
```diff
@@ -1,52 +1,45 @@
...
-
-## Code Splitting
+## Analyzing Code Portions
 
-Commands can be run on a single portion of the code using the `--ls`
-and `--le` options. The former represents the starting line of the code to be
-considered, while the latter its ending line.
-For example, if we want to print the AST of a single function which starts at
-line 5 and ends at line 10, we need to launch this command:
+To analyze only a specific part of the code, use the `--ls` (line start) and `--le` (line end) options:
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:39]:
```suggestion
To analyze only a specific part of the code, use the `--ls` (line start) and `--le` (line end) options.
For example, if we want to print the AST of a single function which starts at line 5 and ends at line 10:
```

File: rust-code-analysis-book/src/commands/nodes.md [hunk 7]
```diff
@@ -1,52 +1,45 @@
...
 
 ```console
 rust-code-analysis-cli -p /path/to/your/file/or/directory -d --ls 5 --le 10
 ```
+
+This prints the AST of the code between lines 5 and 10, useful for analyzing specific functions or blocks.
```
Comment (marco-c) [inline rust-code-analysis-book/src/commands/nodes.md:45]: Remove this one and keep the previous "For example ..." as suggested

Comment (miguelcsx) [reply rust-code-analysis-book/src/commands/nodes.md]: @marco-c I commited your changes, let me know if anything else is missing :+1:

**Review by marco-c — APPROVED** [review]
Thanks!
